plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -328,18 +328,24 @@ func (w *windowsEventLog) getRecord(evtHandle EvtHandle) (*windowsEventLogRecord
 		return nil, fmt.Errorf("utf16ToUTF8Bytes() err %v", err)
 	}
 
+	// The insertion strings could be in either EventData or UserData
```

Comment:
Instead of using `insertPlaceholderValues()`, can you instead try changing the way we call `EvtFormatMessage()`?
It looks like it will find the place holders and replace them with insertion strings (which we can specify).

Look at the `values` parameter :
```
func EvtFormatMessage(publisherMetadata EvtHandle, event EvtHandle, messageID uint32, valueCount uint32, values uintptr, flags EvtFormatMessageFlag, bufferSize uint32, buffer *byte, bufferUsed *uint32)
```

more details:
https://learn.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtformatmessage
>An array of insertion values to use when formatting the event's message string. Typically, you set this parameter to NULL and the function gets the insertion values from the event data itself. You would use this parameter to override the default behavior and supply the insertion values to use. For example, you might use this parameter if you wanted to resolve a SID to a principal name before inserting the value.

Reply:
Maybe it isn't worth the extra effort of doing this.
You've already got some simple code that works.
And I'd hate to go through the extra work just to have Microsoft break that part of the API as well ;)
I'll wait to hear what others think.

Reply:
Fun enough that I did try to transform `EventData`/`UserData` to `dataValues` and `values` at the beginning under the mindset that I should leverage the Microsoft API more. But it didn't work as expected and `EvtFormatMessage` didn't return any error message that I can further fix/fine tune the implementation. 

Also wanna echo you last point that, this way we will not be affected by possible changes to the API. Maybe I can add in the comment that we can later do more research on how to use `EvtFormatMessage` with `dataValues` and `values` and compare both

Reply:
If what we have right now covers all of our bases, I'd rather keep it than pivot.

Reply:
Agree on keeping this as-is and not pivoting. 

I recommend adding a comment near the call to EvtFormatMessage, just saying _we tried_ passing EventData/userData into `valueCount` and `values` parameters, but it did not work. Basically explaining why we chose to implement our own `insertPlaceholderValues`.

Alternatively just include this information in the commit message(s).

That way in a year I won't spend time wondering why we didn't just let `EvtFormatMessage()` do the formatting.

Reply:
Agreed. I'm adding a comment in the next revision.